pkg/scheduler/core/generic_scheduler.go

```diff
@@ -203,27 +205,60 @@ func (g *genericScheduler) findNodesThatFitPod(ctx context.Context, fwk framewor
 
 	// Run "prefilter" plugins.
 	s := fwk.RunPreFilterPlugins(ctx, state, pod)
+	allNodes, err := g.nodeInfoSnapshot.NodeInfos().List()
```

Comment:
Is there any unit test we can add for the changes in this file?

Reply:
The overall logic has been verified by testcase `TestPreferNominatedNode` created in "test/integration/scheduler/preemption_test.go". 
- If the nominated passed all the filters,  the nominated node will be return, otherwise,
- other nodes will be evaluated and return a suitable candidate.

"TestPreferNominatedNode" is truly has not verify the logic of preemption, if you think it's better to test in "generic_scheduler_test.go" or there is any specific change haven't covered by any testcase, please just let me know, I will take care it.

Reply:
hmm, that's integration test, as to the unit test for the change like this, output should be the same before and after the change, it is hard to see which aspect should to be test in the unit test.

Reply:
I like the filter call counts :)